src/main/java/ch/njol/skript/events/EvtDamage.java

```diff
@@ -39,34 +39,51 @@
 @SuppressWarnings("unchecked")
 public class EvtDamage extends SkriptEvent {
 	static {
-		Skript.registerEvent("Damage", EvtDamage.class, EntityDamageEvent.class, "damag(e|ing) [of %entitydata%]")
+		Skript.registerEvent("Damage", EvtDamage.class, EntityDamageEvent.class, "damag(e|ing) [of %entitydata%] [by %entitydata%]")
 				.description("Called when an entity receives damage, e.g. by an attack from another entity, lava, fire, drowning, fall, suffocation, etc.")
-				.examples("on damage:", "on damage of a player:")
-				.since("1.0");
+				.examples("on damage:", "on damage of a player:", "on damage of player by zombie:")
+				.since("1.0, INSERT VERSION for by entitydata");
 	}
 	
 	@Nullable
-	private Literal<EntityData<?>> types;
+	private Literal<EntityData<?>> ofTypes, byTypes;
 	
 	@Override
 	public boolean init(final Literal<?>[] args, final int matchedPattern, final ParseResult parser) {
-		types = (Literal<EntityData<?>>) args[0];
+		ofTypes = (Literal<EntityData<?>>) args[0];
+		byTypes = (Literal<EntityData<?>>) args[1];
 		return true;
 	}
 	
 	@Override
 	public boolean check(final Event evt) {
 		final EntityDamageEvent e = (EntityDamageEvent) evt;
-		if (!checkType(e.getEntity()))
+		if (evt instanceof EntityDamageByEntityEvent) {
+			EntityDamageByEntityEvent event = (EntityDamageByEntityEvent) evt;
+			if (!checkDamager(event.getDamager()))
+				return false;
+		}
```

Comment:
I think you need to check here if byTypes is **not** null

If it isn't null, and if the entity wasn't damaged by another entity, the event shouldn't run

Reply:
I check that in the checkDamager method

Reply:
I mean that if the event isn't an EntityDamageByEntityEvent and byTypes isn't null, you should return false

Reply:
ooooh I see what you mean haha, shall do that ^-^